File: 백준/Bronze/4564. 숫자 카드놀이/숫자 카드놀이.py
from sys import stdin

input = stdin.readline


def get_next_number(num):
    st = list(map(int, str(num)))

    next_number = 1
    for elem in st:
        next_number *= elem
    return next_number


while True:
    num = int(input().rstrip())
    if num == 0:
        break

    while True:
        print(num, end=" ")
        if num // 10 == 0:
            break
        num = get_next_number(num)

File: Java/백준/Bronze/4564. 숫자 카드놀이/숫자 카드놀이.java
import java.util.*;
import java.io.*;

public class Main {

    static BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    static BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
    static StringTokenizer st;

    private static int getNextNumber(int curr) {
        int nextNumber = 1;
        while (curr > 0) {
            nextNumber *= (curr % 10);
            curr /= 10;
        }

        return nextNumber;
    }

    public static void main(String[] args) throws IOException {

        int num;
        while (true) {
            st = new StringTokenizer(br.readLine());
            num = Integer.parseInt(st.nextToken());
            if (num == 0) break;

            while (true) {
                bw.write(num + " ");
                if (num < 10) break;
                num = getNextNumber(num);
            }

            bw.write("\n");
        }

        bw.close();
    }
}